src/main/java/cubicchunks/util/FastCubeBlockAccess.java

```diff
@@ -200,15 +186,21 @@ private IBlockState getBlockState(int blockX, int blockY, int blockZ) {
             } else {
                 ebs.setBlockLight(localX, localY, localZ, val);
             }
-            return;
+            return true;
         }
         Cube cube = getCube(pos.getX(), pos.getY(), pos.getZ());
-        cube.setLightFor(lightType, pos, val);
-        setStorage(pos.getX(), pos.getY(), pos.getZ(), cube.getStorage());
+        if (cube != null) {
+            cube.setLightFor(lightType, pos, val);
+            setStorage(pos.getX(), pos.getY(), pos.getZ(), cube.getStorage());
+            return true;
+        }
+        return false;
     }
 
     @Override public boolean canSeeSky(BlockPos pos) {
         Cube cube = getCube(pos.getX(), pos.getY(), pos.getZ());
+        if (cube == null)
+            return true;
```

Comment:
That's not exactly true. Instead you should have separate array for columns and get column directly bu x/z coords